internal/pkg/service/common/etcdop/iterator/option.go

```diff
@@ -92,9 +96,40 @@ func WithFromSameRev(v bool) Option {
 	}
 }
 
-// WithEnd defines end of the iteration, all keys from the range [prefix/, prefix/end) will be loaded.
-func WithEnd(v string) Option {
+// WithStartOffset defines start of the iteration.
+// An empty string (default) means that the start is the first key in the prefix.
+// Iterated are all keys from the range (prefix/startOffset, prefix/endOffset).
+// Boundary values, start and end offsets, are excluded.
+func WithStartOffset(v string) Option {
 	return func(c *config) {
-		c.end = c.prefix + v
+		c.startOffset = v
 	}
 }
+
+// WithEndOffset defines end of the iteration.
+// An empty string (default) means that the end is the last key in the prefix.
+// Iterated are all keys from the range (prefix/startOffset, prefix/endOffset).
+// Boundary values, start and end offsets, are excluded.
+func WithEndOffset(v string) Option {
+	return func(c *config) {
+		c.endOffset = v
+	}
+}
+
+func (c config) start() string {
+	if c.startOffset != "" {
+		// Iterate from the startOffset, the startOffset is excluded.
+		return etcd.GetPrefixRangeEnd(c.prefix + c.startOffset)
```

Comment:
`GetPrefixRangeEnd` is a helper to get next key after the specified key, ... it adds (lexicographically) one character to the key.

By default, the start of an etcd range query is included in results, so we are using the helper to exclude it.

Reply:
Yes, it is well covered:

https://github.com/keboola/keboola-as-code/blob/michaljurecko-PSGO-287-list-source-e2e/internal/pkg/service/common/etcdop/iterator/iterator_test.go#L41 -> `iterator.WithStartOffset`

![image](https://github.com/keboola/keboola-as-code/assets/19371734/8fdf6c3b-5b4b-458e-9b3f-7cc14e32eea4)